File: java/clase12/Ventas/src/ar/com/cibergauchos/ventas/Orden.java
package ar.com.cibergauchos.ventas;

public class Orden {

    private int idOrden;
    private Producto productos[];
    private static int contadorOrdenes;
    private int contadorProductos;
    private static final int MAX_PRODUCTOS = 10;

    public Orden() {
        this.idOrden = ++Orden.contadorOrdenes;
        this.productos = new Producto[Orden.MAX_PRODUCTOS];
    }

    public void agregarProducto(Producto producto) {
        if (this.contadorProductos < Orden.MAX_PRODUCTOS) {
            this.productos[this.contadorProductos++] = producto;
        } else {
            System.out.println("Se ha superado el máximo de productos: " + Orden.MAX_PRODUCTOS);
        }
    }

    public double calcularTotal() {
        double total = 0;
        for (int i = 0; i < this.contadorProductos; i++) {
            total += this.productos[i].getPrecio();
        }
        return total;
    }

    public void mostrarOrden() {
        System.out.println("------------------------ Orden n° " + this.idOrden + " ------------------------");
        double totalOrden = this.calcularTotal();
        System.out.println("## Total: $" + totalOrden);
        System.out.println("## Productos: ");
        for (int i = 0; i < this.contadorProductos; i++) {
            System.out.println("  *"+this.productos[i]);
        }
    }
}

File: python/clase11/DiseñoClasesYsobrecargaOperadores/Orden.py
from Producto import Producto

class Orden:
    contador_ordenes = 0

    def __init__(self, productos):
        Orden.contador_ordenes += 1
        self._id_orden = Orden.contador_ordenes
        self._productos = list(productos)

    def agregar_producto(self, producto):
        self._productos.append(producto) #Para agregar un nuevo producto

    def calcular_total(self):
        total = 0 # Variable temporal para almacenar el total de la orden
        for producto in self._productos:
            total += producto.precio
        return total
    
    def __str__(self):
        productos_str = ""
        for producto in self._productos:
            productos_str += producto.__str__() + " | "
        return f'Orden: {self._id_orden}, \nProductos: {productos_str}'

if __name__ == "__main__":
    producto1 = Producto("Camiseta", 1000)
    producto2 = Producto("Pantalon", 150)
    productos1 = [producto1, producto2] # Lista de productos
    orden1 = Orden(productos1) # Creamos una orden con los productos(lista pasada)
    print(orden1)

    producto3 = Producto("Medias", 50)
    producto4 = Producto("Zapatos", 700)
    productos2 = [producto3, producto4] # Lista de productos
    orden2 = Orden(productos2) # Creamos una orden con los productos(lista pasada)
    print(orden2)